src/test/java/ejercicio/banco/repository/MySqlBankRepositoryTest.java

```diff
@@ -8,90 +8,110 @@
 
     @Test
-    public void testFindAllReturnsFullList() {
-        full.add(expectedBank1);
-        full.add(expectedBank2);
-        banks = repository.findAll(TABLE);
-        assertEquals(full.size(), banks.size());
-    }
+    public void testFindReturnsNullWhenItemNotFound() throws SQLException {
+        executeQuery(util, "select * from BANKS");
 
+        when(resultSet.next()).thenReturn(true).thenReturn(false);
+        when(resultSet.getString("name")).thenReturn(expectedBank2.getName());
+        when(resultSet.getInt("id")).thenReturn(expectedBank2.getId());
+        when(resultSet.getString("address")).thenReturn(expectedBank2.getAddress());
 
-    @Test
-    public void testFindReturnsNullWhenItemNotFound() {
         bank = repository.find(3);
-        assertEquals(bank, null);
+        assertNull(bank);
     }
 
     @Test
-    public void testFindReturnsProperItem() {
+    public void testFindReturnsProperItem() throws SQLException {
+        executeQuery(util, "select * from BANKS");
+
+        when(resultSet.next()).thenReturn(true).thenReturn(false);
+        when(resultSet.getString("name")).thenReturn(expectedBank2.getName());
+        when(resultSet.getInt("id")).thenReturn(expectedBank2.getId());
+        when(resultSet.getString("address")).thenReturn(expectedBank2.getAddress());
+
         bank = repository.find(2);
         assertEquals(bank.getName(), expectedBank2.getName());
     }
 
     @Test
-    public void testFindByNameThrowsExceptionWhenNull() {
+    public void testFindByNameThrowsExceptionWhenNull() throws SQLException {
+        executeQuery(util, "select * from BANKS");
+
+        when(resultSet.next()).thenReturn(true).thenReturn(false);
+        when(resultSet.getString("name")).thenReturn(expectedBank2.getName());
+        when(resultSet.getInt("id")).thenReturn(expectedBank2.getId());
+        when(resultSet.getString("address")).thenReturn(expectedBank2.getAddress());
+
+        List<Bank> banks;
         banks = repository.findByName(null, "name");
-        assertEquals(empty, banks);
+        assertTrue(banks.isEmpty());
     }
 
     @Test
-    public void testFindByNameThrowsExceptionWhenListIsEmpty() {
-        banks = repository.findByName(EMPTYTABLE, "name");
-        assertEquals(empty, banks);
-    }
+    public void testFindByNameReturnsNullWhenItemsNotFound() throws SQLException {
+        executeQuery(util, "select * from BANKS");
 
-    @Test
-    public void testFindByNameReturnsNullWhenItemsNotFound() {
-        banks = repository.findByName(TABLE, "nothing");
-        assertEquals(empty, banks);
+        when(resultSet.next()).thenReturn(true).thenReturn(false);
+        when(resultSet.getString("name")).thenReturn(expectedBank2.getName());
+        when(resultSet.getInt("id")).thenReturn(expectedBank2.getId());
+        when(resultSet.getString("address")).thenReturn(expectedBank2.getAddress());
+
+        List<Bank> banks;
+        banks = repository.findByName("", "nothing");
```

Comment:
This two lines should be done in one line:
```
List<Bank> banks = repository.findByName("", "nothing");
```

Reply:
Done